### PR TITLE
Add a note about trailing NUL characters in int32.TryParse

### PR DESCRIPTION
**Pull request by Porges**
This is addressing https://github.com/dotnet/corefx/issues/29405.

If this text is okay I'll go ahead and add it to the other numeric parsing docs.

### CODE REVIEW

File: xml/System/Int32.xml [hunk 1]
```diff
@@ -2423,6 +2423,8 @@
 |*e*|The 'e' or 'E' character, which indicates that the value is represented in exponential notation. The `s` parameter can represent a number in exponential notation if `style` includes the <xref:System.Globalization.NumberStyles.AllowExponent?displayProperty=nameWithType> flag.|  
 |*hexdigits*|A sequence of hexadecimal digits from 0 through f, or 0 through F.|  
   
+ Note that, for compatibility reasons, a single trailing NUL (U+0000) character is always permitted in `s`, regardless of the value of `style`.
```
Comment (rpetrusha) [inline xml/System/Int32.xml]: I think that I'd do this as a note and phrase it somewhat differently:
\> [!NOTE]
\> Any terminating NUL (U+0000) characters in \`s\` are ignored by the parsing operation regardless of the value of the \`style\` argument.

**Comment by Porges** [comment]
> Unless I'm mistaken, any number of terminating nulls are ignored

This is correct, I misremembered. Will update the text and change to a real "note" as you suggested.

**Comment by rpetrusha** [comment]
@Porges, would you like us to make the changes for you?

**Comment by Porges** [comment]
@rpetrusha I should be able to tackle it this weekend – I just moved houses/countries so I haven't had a lot of time 🙂

**Comment by Porges** [comment]
... or now. I've changed to the suggested text and added it to the other locations.

**Comment by rpetrusha** [comment]
Thank you so much for doing this, @Porges. This looks really good, except that the note has to be on two lines, like this:

> \> [!NOTE] 
> \> Any terminating NUL (U+0000) characters in `s` are...
> 

Would you like to make the change, or would you like for us to make it for you? Once that's done, we can merge.

**Comment by Porges** [comment]
There we go, I had to fight my markdown editor a bit because it keeps wanting to reflow the text 🙂

**Comment by rpetrusha** [comment]
Thank you, @Porges, for adding the note to all the relevant places, and then for changing it to a multi-line note. This looks really good, so I'll merge your PR now. The changes should appear live on docs.microsoft.com in the next day or two.

**Comment by Porges** [comment]
Thanks! ☺️

**Pull request merged**